api/tests/test_asset_type_model.py

```diff
@@ -1,26 +1,30 @@
 from django.test import TestCase
 
-from ..models import AssetType
+from ..models import AssetType, AssetSubCategory
 
 
 class AssetTypeModelTest(TestCase):
     """Tests for the Asset Types Model"""
 
     def setUp(self):
-        test_asset = AssetType(asset_type="TestAsset")
-        test_asset.save()
+        self.new_sub_category = AssetSubCategory(
+            sub_category_name="Computer Assessories")
+        self.new_sub_category.save()
+        AssetType.objects.create(asset_type="Headset",
+                                 asset_sub_category=self.new_sub_category)
         self.all_assettypes = AssetType.objects.all()
 
     def test_add_new_asset_type(self):
```

Comment:
can we also check that a type isn't created if not associated with a sub-category?

Reply:
I think it is but isn't that trying to test Django's code ? I tried testing this by adding a type without adding an existing subcategory and an error is thrown. This is okay but capturing the error and asserting it is proving difficult.

Reply:
which error are you getting?

Reply:
I managed to have a test for this actually. I implemented with the `self.assertRaises`

Reply:
awesome!